File: bowling/bowling.py
class BowlingGame():
    
    def __init__(self):
        self.rolls = []

    def roll(self, pins):
        self.rolls.append(pins)

    def isSpare(self, roll):
        return (self.rolls[roll] + self.rolls[roll + 1] == 10)
    
    def isStrike(self, roll):
        return self.rolls[roll] == 10

    def spareBonus(self, roll):
        return 10 + self.rolls[roll + 2]

    def strikeBonus(self, roll):
        return 10 + self.rolls[roll + 1] + self.rolls[roll + 2]

    def score(self):
        scoreTotal = 0
        roll = 0
        for frame in range(10):
            if self.isStrike(roll):
                scoreTotal += self.strikeBonus(roll)
                roll += 1
            elif self.isSpare(roll):
                scoreTotal += self.spareBonus(roll)
                roll += 2
            else:
                scoreTotal += (self.rolls[roll] + self.rolls[roll + 1])
                roll += 2
        return scoreTotal;

File: php/src/Bowling.php
<?php

namespace PhpKata180627;

class Bowling
{
    private $rolls = array();
    
    public function roll($pins)
    {
        $this->rolls[] = $pins;
    }

    public function score()
    {
        $score = 0;
        $rollIndex = 0;
        for ($frame = 1; $frame <= 10; $frame++) {
            if ($this->isStrike($rollIndex)) {
                $score += (10 + $this->strikeBonus($rollIndex));
                $rollIndex += 1;
            } elseif ($this->isSpare($rollIndex)) {
                $score += (10 + $this->spareBonus($rollIndex));
                $rollIndex += 2;
            } else {
                $score += $this->sumOfScoreInFrame($rollIndex);
                $rollIndex += 2;
            }
        }
        return $score;
    }

    private function isStrike($rollIndex)
    {
        return 10 == $this->rolls[$rollIndex];
    }

    private function isSpare($rollIndex)
    {
        return 10 == ($this->rolls[$rollIndex] + $this->rolls[$rollIndex + 1]);
    }

    private function strikeBonus($rollIndex)
    {
        return $this->rolls[$rollIndex + 1] + $this->rolls[$rollIndex + 2];
    }

    private function spareBonus($rollIndex)
    {
        return $this->rolls[$rollIndex + 2];
    }

    private function sumOfScoreInFrame($rollIndex)
    {
        return $this->rolls[$rollIndex] + $this->rolls[$rollIndex + 1];
    }
}
